### PR TITLE
NotifierKafka: avoid timeout case when there are no persist messages and using OffsetOldest or offsetDuration

### PR DESCRIPTION
**Pull request by fkaleo**

### CODE REVIEW

File: mdata/notifierKafka/notifierKafka.go [hunk 1]
```diff
@@ -82,6 +82,10 @@ func (c *NotifierKafka) start() {
 		if err != nil {
 			log.Fatalf("kafka-cluster: failed to get offset %d: %s", offsetTime, err)
 		}
+		if startOffset < 0 {
+			// happens when OffsetOldest or an offsetDuration was used and there is no message in the partition
```
Comment (Dieterbe) [inline mdata/notifierKafka/notifierKafka.go]: is this still accurate? AFAICT startOffset is the concrete (positive) offset, whereas offsetTime is now the "conceptual" (possibly negative) offset

Comment (woodsaj) [reply mdata/notifierKafka/notifierKafka.go]: yes.   The comment clearly explains why this check is needed, ie `c.client.GetOffset()` will return '-1' if there are no messages in the partition.

Comment (fkaleo) [reply mdata/notifierKafka/notifierKafka.go]: yeah the idea is that the concrete offset (startOffset) is not absolutely always positive as I imagined. It is negative in the specific case no message is in the queue.

Comment (Dieterbe) [reply mdata/notifierKafka/notifierKafka.go]: what if it was set to newest and there are no messages in the partition?

Comment (fkaleo) [reply mdata/notifierKafka/notifierKafka.go]: In that case startOffset is set to 0

**Pull request merged**